src/librustc_mir/build/scope.rs

```diff
@@ -630,17 +630,20 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                          span: Span,
                          region_scope: region::Scope,
                          place: &Place<'tcx>,
-                         place_ty: Ty<'tcx>) {
+                         place_ty: Ty<'tcx>,
+                         drop_kind: DropKind) {
         let needs_drop = self.hir.needs_drop(place_ty);
-        let drop_kind = if needs_drop {
-            DropKind::Value { cached_block: CachedBlock::default() }
-        } else {
-            // Only temps and vars need their storage dead.
+
+        if let DropKind::Storage = drop_kind {
```

Comment:
I'd suggest using a `match` and making both variants explicit, with `Value` first (just to make the diff more readable).